### PR TITLE
[mb2] Add the 'scrape' command. JB#59443

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: sdk-setup/src/mb2 [hunk 1]
```diff
@@ -2703,7 +2712,7 @@ init_patch_wrapper() (
         local untracked_added=()
         readarray -t untracked_added < <(join -v 2 \
             <(printf '%s' "$untracked_old") <(printf '%s' "$untracked_now"))
-        if [[ $untracked_added ]]; then
+        if [[ ${#untracked_added[*]} -gt 0 ]]; then
```
Comment (vigejolla) [inline sdk-setup/src/mb2:2761]: How is this related to -o nounset?

File: sdk-setup/src/mb2 [hunk 2]
```diff
@@ -531,9 +532,54 @@ COMMANDS
...
+        (superproject). This command automates the job of collecting patches
+        for new commits appearing in all (sub-)submodules and updating
+        existing patch files in a safe manner.
+
+        This command may be used in pair with the 'prepare' command, which is
+        capable of doing the opposite job: apply patches as commits (unless
+        the '--no-vcs-apply' global option is used).
+
+        This command enforces certain patch formatting and file naming
+        conventions. It is not guarranteed (not desired) that it produces
```
Comment (vigejolla) [inline sdk-setup/src/mb2:564]: extra 'r' in guaranteed

**Review by vigejolla — APPROVED** [review]
One question, which I would like answered, and one typo. Approving already; If you make no further changes besides fixing the typo, feel free to merge.

File: sdk-setup/src/mb2 (hunk 1 above)
Comment (martyone) [reply sdk-setup/src/mb2:2761]: The original code fails with empty array (or more exactly an array with item at index 0 unset).

File: sdk-setup/src/mb2 (hunk 2 above)
Comment (martyone) [reply sdk-setup/src/mb2]: I really hate this word :) fixed.

**Comment by martyone** [comment]
Polished more

**Review by vigejolla — APPROVED** [review]
Still LGTM

**Comment by martyone** [comment]
Extended it to remove the commits after creating patches from them, renamed the command to avoid any doubt on which command is meant (whether mb2's subcommand or git's subcommand), moved it to it's own domain so that it does not disturb the row of the closely related commands prep, build, qmake, make, package etc.

**Review by vigejolla — APPROVED** [review]
It just keeps getting better :)
